ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/PortedEndpoints.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import static io.confluent.ksql.rest.Errors.toErrorCode;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableSet;
+import io.confluent.ksql.api.auth.ApiSecurityContext;
+import io.confluent.ksql.api.auth.DefaultApiSecurityContext;
+import io.confluent.ksql.api.spi.EndpointResponse;
+import io.confluent.ksql.api.spi.Endpoints;
+import io.confluent.ksql.json.JsonMapper;
+import io.confluent.ksql.rest.entity.ClusterTerminateRequest;
+import io.confluent.ksql.rest.entity.KsqlErrorMessage;
+import io.confluent.ksql.rest.entity.KsqlRequest;
+import io.confluent.ksql.rest.entity.Versions;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiFunction;
+import javax.ws.rs.core.MediaType;
+
+public class PortedEndpoints {
+
+  private static final Set<String> PORTED_ENDPOINTS = ImmutableSet.of("/ksql");
+
+  private final Endpoints endpoints;
+  private final Server server;
+
+  public PortedEndpoints(final Endpoints endpoints, final Server server) {
+    this.endpoints = endpoints;
+    this.server = server;
```

Comment:
nit: validate params that will be stored in object state; ensuring object does not get into an invalid state.

Reply:
ack